packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -33,8 +32,20 @@ const bar = new ProgressBar(
   }
 )
 
+const reportError = (message, err, reporter) => {
+  if (reporter) {
+    reporter.error(message, err)
+  } else {
+    console.error(message, err)
+  }
+
+  if (process.env.gatsby_executing_command === `build`) {
```

Comment:
this is already so specific to gatsby, I don't think there is any harm in requiring the reporter to be passed to gatsby-plugin-sharp, I guess maybe it's for backwards compat though ?

Reply:
Initially I wanted to backport `panicOnBuild` addition to reporter in `gatsby-cli` from v2 (which this is copied from), but thought about unofficial plugins that might be slower to update (and pass reporter) so used this as bandaid solution instead of just breaking not updated plugins.

Reply:
Why not just split out the reporting to a separate package that can be used anywhere?

I'll merge and release this PR in the meantime. But no reason to duplicate functionality. There's nothing tying the reporting to gatsby-cli.

Reply:
My concern wasn't the package sharing, but the singleton. Eventually to do reporting well coordinated across plugins we need more explicit control over how plugins report, since grouping output from a a bunch of different sources in parallel is tough

Reply:
But NPM would dedupe the installation so there should be just one version of the reporter package that's running right?

Reply:
sorry yeah, but waht i'm saying you might not _want_ a singleton, you may wnat to pass a "sub reporter" instance or something per plugin that only has functionality to print to it's group for instance. I think i'd need to try and demonstrate what i mean, i had made an attempt before but got sidetracked a bit.

Reply:
Ah gotcha.

Hmmm we could still do both right? Normal plugins would depend on the reporter passed and and the few oddballs like gatsby-plugin-sharp could depend on it directly.